src/rust/grapl-web-ui/src/routes/api/plugin/create.rs

```diff
@@ -0,0 +1,172 @@
+        plugin_id: plugin_id,
+    }))
+}
+
+async fn get_metadata(payload: &mut Multipart) -> Result<CreateParametersMetadata, PluginError> {
+    if let Some(mut field) = payload.try_next().await? {
+        let content_disposition = field.content_disposition();
+        let field_name = content_disposition
+            .get_name()
+            .ok_or(PluginError::BadRequest {
+                message: "multipart/form-data name missing".to_string(),
+            })?;
+
+        let metadata_field_name = "metadata";
+
+        if field_name != metadata_field_name {
+            return Err(PluginError::UnexpectedPart {
+                expected: metadata_field_name,
+                found: field_name.to_string(),
+            });
+        }
+
+        // Get form body
+        let mut body = web::BytesMut::new();
+        if let Some(chunk) = field.next().await {
+            let chunk = chunk?;
+
+            // TODO: Reintroduce this when we can stream plugin artifact upstream
+            // // limit max size of in-memory payload
+            // if (body.len() + chunk.len()) > MAX_SIZE {
+            //     return Err(WebResponseError::MultipartChunkTooLarge {
+            //         size: chunk.len(),
+            //         max_size: MAX_SIZE,
+            //     })
+            // }
+
+            body.extend_from_slice(&chunk);
+        }
+
+        let metadata = serde_json::from_slice::<CreateParametersMetadata>(&body)?;
+
+        return Ok(metadata);
+    }
+
+    Err(PluginError::BadRequest {
+        message: "missing form data for plugin/create metadata".to_string(),
+    })
+}
+
+// TODO: this almost copy pasta from get_metadata but it won't be once we can stream plugin artifact upstream
+// without first loading them all into memory first.
+async fn get_plugin_artifact(payload: &mut Multipart) -> Result<web::Bytes, PluginError> {
+    if let Some(mut field) = payload.try_next().await? {
+        let content_disposition = field.content_disposition();
+        let field_name = content_disposition
+            .get_name()
+            .ok_or(PluginError::BadRequest {
+                message: "multipart/form-data name missing".to_string(),
+            })?;
+
+        let metadata_field_name = "plugin_artifact";
+
+        if field_name != metadata_field_name {
+            return Err(PluginError::UnexpectedPart {
+                expected: metadata_field_name,
+                found: field_name.to_string(),
+            });
+        }
+
+        // Get form body
+        let mut body = web::BytesMut::new();
+        if let Some(chunk) = field.next().await {
+            let chunk = chunk?;
+            body.extend_from_slice(&chunk);
+        }
+
+        return Ok(body.into());
+    }
+
+    Err(PluginError::BadRequest {
+        message: "missing form data for plugin/create plugin artifact".to_string(),
+    })
+}
+
+// It'd be greate if we didn't have to do this and PluginType jsut derived serde::Deserialize itself
```

Comment:
thats a solid point; we should bring it up w jesse. all our enums are not currently SerDe